plugin/pkg/admission/noderestriction/admission.go

```diff
@@ -677,11 +681,16 @@ func (p *Plugin) podReferencesAudience(pod *v1.Pod, audience string) (bool, erro
 		switch {
 		case v.Ephemeral != nil && v.Ephemeral.VolumeClaimTemplate != nil:
 			pvcName := ephemeral.VolumeClaimName(pod, &v)
-			driverName, err = p.getCSIFromPVC(pod.Namespace, pvcName)
+			driverName, err = p.getCSIFromPVC(ctx, pod.Namespace, pvcName)
 		case v.PersistentVolumeClaim != nil:
-			driverName, err = p.getCSIFromPVC(pod.Namespace, v.PersistentVolumeClaim.ClaimName)
+			driverName, err = p.getCSIFromPVC(ctx, pod.Namespace, v.PersistentVolumeClaim.ClaimName)
 		case v.CSI != nil:
 			driverName = v.CSI.Driver
+		default:
```

Comment:
It is better to check if the volume is inline migratable before trying to get csi driver name.
```go
if !p.csiTranslator.IsInlineMigratable(&v) {
   continue
}
```

Reply:
```
	foundAudiencesInPodSpec, err := p.podReferencesAudience(pod, requestedAudience)
	if err != nil {
		return fmt.Errorf("error validating audience %q: %w", requestedAudience, err)
	}
	if !foundAudiencesInPodSpec {
		return fmt.Errorf("audience %q not found in pod spec volume", requestedAudience)
	}
```

the message `audience %q not found in pod spec volume` is expected if a pod has a volume but the volume is unmigratable, i.e. configMap, secret, downwardAPI, projected, etc.

Reply:
> It is better to check if the volume is inline migratable before trying to get csi driver name.

thanks, updated!

> the message audience %q not found in pod spec volume is expected if a pod has a volume but the volume is unmigratable, i.e. configMap, secret, downwardAPI, projected, etc.

not sure about what this comment is saying? that error message indicates there is no volume source that has this audience specified, so kubelet shouldn't be allowed to request a token for this audience.